File: ios/LoverApp/Services/SupabaseClient.swift
import Foundation
import Supabase

// Singleton wrapper around the Supabase Swift SDK client. Used by AuthService
// for sign-in and (Phase 3b+) by PairingService and (Phase 4+) by ChatService.
//
// Usage:
//   try await SB.client.auth.signInWithIdToken(...)
//   try await SB.client.from("users").select().execute()

enum SB {
    static let client = SupabaseClient(
        supabaseURL: SupabaseConfig.url,
        supabaseKey: SupabaseConfig.anonKey
    )
}

File: android/app/src/main/kotlin/michel/kit/us/data/SupabaseClient.kt
package michel.kit.us.data

import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.postgrest.Postgrest
import io.github.jan.supabase.realtime.Realtime
import io.github.jan.supabase.storage.Storage
import kotlinx.serialization.json.Json

// Singleton wrapper around the Supabase Kotlin SDK client. Mirror of
// ios/LoverApp/Services/SupabaseClient.swift (the `SB` enum).
//
// Usage:
//   SupabaseClient.instance.auth.signInWith(...)
//   SupabaseClient.instance.postgrest["messages"].select { ... }
object SupabaseClient {
    val instance by lazy {
        createSupabaseClient(
            supabaseUrl = SupabaseConfig.URL,
            supabaseKey = SupabaseConfig.ANON_KEY
        ) {
            install(Auth)
            install(Postgrest)
            install(Realtime)
            install(Storage)
            defaultSerializer = io.github.jan.supabase.serializer.KotlinXSerializer(
                Json {
                    ignoreUnknownKeys = true
                    explicitNulls = false
                    encodeDefaults = true
                }
            )
        }
    }
}
